classes/observer.php

```diff
@@ -135,4 +137,334 @@ public static function mod_quiz_event_attempt_submitted(
+        $cl = new \plagiarism_copyleaks_comms();
+        $commentdata = (array)[
+            'commentId' => $eventdata['objectid'],
+            'courseModuleId' => $eventdata['contextinstanceid'],
+            'moodleUserId' => $eventdata['userid'],
+        ];
+
+        $submissionid = $eventdata['other']['itemid'];
+
+        $cl->delete_assign_submission_comment($commentdata, $submissionid);
+    }
+
+    /**
+     * user graded event handler.
+     * @param \core\event\user_graded $event
+     */
+    public static function user_graded(
+        \core\event\user_graded $event
+    ) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+        // if (!\plagiarism_copyleaks_moduleconfig::is_module_enabled($this->task->get_modulename(), $newcmid)) {
+        //     return;
+        // }
+        global $DB;
+        $eventdata = $event->get_data();
+
+        if (!$item = $DB->get_record('grade_items', ['id' => $eventdata['other']['itemid']])) {
+            return;
+        }
+
+        if (
+            $item->itemtype == 'mod'
+        ) {
+
+            $cl = new \plagiarism_copyleaks_comms();
+
+            if (!$DB->get_record('grade_grades', ['id' => $eventdata['objectid']])) {
+                return;
+            }
+
+            $cmid = get_coursemodule_from_instance($item->itemmodule, $item->iteminstance, $eventdata['courseid'])->id;
+            $userid     = ($eventdata['relateduserid']) ? $eventdata['relateduserid'] : $eventdata['userid'];
+            $finalgrade = $eventdata['other']['finalgrade'];
+
+            $data = (array)[
+                'courseModuleId' => $cmid,
+                'moodleUserId' => $userid,
+                'finalGrade' => $finalgrade,
+            ];
+
+            $cl->upsert_assign_grade($data);
+        }
+    }
+
+    /**
+     * Group created event handler.
+     * @param \core\event\group_created $event
+     */
+    public static function group_created(\core\event\group_created $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+        
+        $eventdata = $event->get_data();
+        $cl = new \plagiarism_copyleaks_comms();
+        $groupname = groups_get_group_name($eventdata['objectid']);
+        $courseId = $eventdata['courseid'];
+        $data = (array)[
+            'groupId' => $eventdata['objectid'],
+            'groupName' => $groupname,
+        ];
+
+        $cl->create_group($data, $courseId);
+    }
+
+    /**
+     * Group deleted event handler.
+     * @param \core\event\group_deleted $event
+     */
+    public static function group_deleted(\core\event\group_deleted $event) {
+        // Check if Copyleaks API is connected.
+        if (!plagiarism_copyleaks_dbutils::is_copyleaks_api_connected()) {
+            return;
+        }
+        
```

Comment:
@ShadeAma shouldn't we check if assign is enabled here? same as other events

if (!plagiarism_copyleaks_moduleconfig::is_module_enabled('assign', $cmid)) {
return;
}